### PR TITLE
Fixed hardware wallet info popup on token allowance screen

### LINKED ISSUE #17486: [Bug]: Approve an ERC20 Token does not display the Hardware Wallet info section while using a Hardware Wallet Account
### Describe the bug

**Problem**: when I am using a Hardware Wallet Account, if I try to Approve an ERC20 token I can see that the Confirmation page does not display the typical Hardware Wallet message, with instructions on signing the tx on my Harware Wallet.

https://user-images.githubusercontent.com/54408225/215451707-c412cec2-ea55-4b46-863a-dc5fea99251e.mp4

**Expected**: see HW info section here as example

Screenshot from 2023-01-30 11-22-24

### Steps to reproduce

1. Import a Hardware Wallet account
2. Go to test dapp https://metamask.github.io/test-dapp/
3. Connect HW Account
4. Click Create Token
5. Accept tx on MM and on Hardware Wallet
6. Click Approve Tokens -- see no HW text appears on Approve and Confirmations screens

### Error messages or log output

_No response_

### Version

10.24.1

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation
* Fixes #17486

### After
https://user-images.githubusercontent.com/92310504/220913281-ee2d9471-4737-4854-807b-cd4249c0285c.mov

## Manual Testing Steps
1. Import a hardware wallet account
2. Go to test dapp - `https://metamask.github.io/test-dapp/`
3. Connect HW account
4. Click Create Token
5. Accept tx on MM and on Hardware Wallet
6. Click Approve Tokens

### CODE REVIEW

**Comment by Sylvia24** [comment]
We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

File: ui/pages/token-allowance/token-allowance.js [hunk 1]
```diff
@@ -491,6 +494,11 @@ export default function TokenAllowance({
           </Box>
         </Box>
       ) : null}
+      {!isFirstPage && isHardwareWalletConnected && (
+        <Box paddingLeft={2} paddingRight={2}>
+          <LedgerInstructionField showDataInstruction />
+        </Box>
+      )}
```
Comment (jpuri) [inline ui/pages/token-allowance/token-allowance.js:501]: Hey will be nice to have test coverage for above.

**Review by digiwand — COMMENTED** [review]
LGTM, I agree with @jpuri that it would be handy to add tests. I can approve after we add a test

File: ui/pages/token-allowance/token-allowance.js (hunk 1 above)
Comment (brad-decker) [reply ui/pages/token-allowance/token-allowance.js:501]: Seconded.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/token-allowance/token-allowance.js (hunk 1 above)
Comment (filipsekulic) [reply ui/pages/token-allowance/token-allowance.js:501]: Added here: ce80c0e8be84def35a07a12f42a8c467bf29500f

File: ui/pages/token-allowance/token-allowance.test.js [hunk 2]
```diff
@@ -245,4 +253,28 @@ describe('TokenAllowancePage', () => {
...
+  it('should show hardware wallet info text', () => {
+    const { queryByText, getByText, getByTestId } = renderWithProvider(
+      <TokenAllowance {...props} />,
+      store,
+    );
+
+    const textField = getByTestId('custom-spending-cap-input');
+    fireEvent.change(textField, { target: { value: '1' } });
+
+    const nextButton = getByText('Next');
```
Comment (digiwand) [inline ui/pages/token-allowance/token-allowance.test.js:266]: nice! tests look good. could be handy to add an additional null check to check the text is not showing on the first page 

```suggestion
    expect(queryByText('Prior to clicking confirm:')).toBeNull();

    const nextButton = getByText('Next');
```

Comment (filipsekulic) [reply ui/pages/token-allowance/token-allowance.test.js:268]: Good suggestion! Thanks!
Changed it here: 455663deac8390e9a234fbf8e4bbcf389508066c

**Review by jpuri — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**